src/utils.h

```diff
@@ -126,3 +126,18 @@ inline int getRandomInt(int min, int max)
 
     return distrib(gen);
 }
+
+inline std::chrono::time_point<std::chrono::high_resolution_clock> highResolutionNow()
+{
+    return std::chrono::high_resolution_clock::now();
+}
+
+inline long double elapsedMicroseconds(std::chrono::time_point<std::chrono::high_resolution_clock> startTime)
+{
+    return std::chrono::duration<long double, std::micro>(highResolutionNow() - startTime).count();
+}
+
+bool fileExists(const char *filename)
+{
+    return GetFileAttributes(filename) != INVALID_FILE_ATTRIBUTES;
+}

```

Comment:
Do we really need this help? Atm we are using [`stat()`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=msvc-170) everywhere as I think it's the fastest call we can make on the OS to check if a file exists. I'd be happy to port all the [`stat()`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=msvc-170) calls into a unique utility function but we need to verify which is the fastest call ( and compatible one with 7th too ).

Reply:
I can use stat, no problem. But is there benchmarks of this? Stat should use GetFilesAttributes in its implementation, so I don't think it is faster.

Reply:
So when I started to use it around, this was my ref thread: https://stackoverflow.com/a/12774387

But looks like we might want to use the native `std::filesystem::exists` way as it's meant to be exactly for that purpose, so we should hope that optimizations done at that level are smart enough to be "the fastest" option.

I'm searching on the `ucrt` source code for what `stat` uses but atm I can't find any trace of `GetFileAttributes` which seems to be used by the [`access()`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/access-waccess?view=msvc-170) API.

Reply:
Your reference does not have a benchmark on Windows. For me "stat()" is a POSIX call, Microsoft implemented it on top of its own syscalls, implying an overhead, with CreateFileW as you found.

Do you think if I use stat in fileExists, I could replace all file exists helpers in FFNx?

Reply:
I think so yes, and it should keep the compatibility with 7th too.

Reply:
One more update: `std::filesystem:exists` seems to use the [`GetFileAttributesExW`](https://github.com/microsoft/STL/blob/main/stl/src/filesystem.cpp#L872) API so this would break 7th compatibility as well. We could adopt it, but we would need to ship a new stable 7th version that hooks this API, test that everything works as expected and then finally be able to ship an FFNx stable round. It is doable but it requires some effort.

So for now going with `stat` is the only safe way to ensure everything we build so far continues to work.